### PR TITLE
Remove `3.x` VR tutorials, add landing page for future XR docs

### PR DESCRIPTION
**Pull request by m4gr3d**

### CODE REVIEW

**Comment by m4gr3d** [comment]
@BastiaanOlij Ping on this PR!

**Comment by m4gr3d** [comment]
@Calinou Here's the 4.x version of https://github.com/godotengine/godot-docs/pull/5480.

**Comment by m4gr3d** [comment]
@Calinou @mhilbrunner Ping on this PR!

File: tutorials/xr/index.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+XR
+==
+
+Work In Progress...
+-------------------
```
Comment (Calinou) [inline tutorials/xr/index.rst:5]:
```suggestion
Work in progress...
-------------------

Documentation for XR in Godot 4.0 XR hasn't been written yet. Please check back in the future.
```

**Review by Calinou — APPROVED** [review]
Thanks!

**Pull request merged**